aks/storage_account/resources.tf

```diff
@@ -0,0 +1,77 @@
+locals {
+  name = var.name != null ? var.name : "default"
+  storage_account_name = var.storage_account_name_override != null ? var.storage_account_name_override : lower(
+    "${var.azure_resource_prefix}${var.service_short}${local.name}${var.config_short}sa"
+  )
+}
+
+data "azurerm_resource_group" "main" {
+  name = "${var.azure_resource_prefix}-${var.service_short}-${var.config_short}-rg"
+}
+
+resource "azurerm_storage_account" "main" {
+  access_tier                       = "Hot"
+  account_kind                      = "StorageV2"
+  account_replication_type          = var.environment != "production" ? "LRS" : var.production_replication_type
+  account_tier                      = "Standard"
+  allow_nested_items_to_be_public   = false
+  https_traffic_only_enabled        = true
+  infrastructure_encryption_enabled = var.infrastructure_encryption_enabled
+  location                          = "UK South"
+  min_tls_version                   = "TLS1_2"
+  name                              = local.storage_account_name
+  public_network_access_enabled     = var.public_network_access_enabled
+  resource_group_name               = data.azurerm_resource_group.main.name
+
+  blob_properties {
```

Comment:
this section would need to be optional I think. As we might need want a delete retention policy for one or both. e.g. review apps we would be happy to delete immediately
Also versioning would need to be optional.